python/catkin/builder.py

```diff
@@ -922,9 +923,15 @@ def build_workspace_isolated(
                 _print_build_error(package, e)
                 # Let users know how to reproduce
                 # First add the cd to the build folder of the package
-                cmd = 'cd ' + os.path.join(buildspace, package.name) + ' && '
+                cmd = 'cd ' + pipes.quote(os.path.join(buildspace, package.name)) + ' && '
```

Comment:
`pipes.quote` is a deprecated function. Please use `shlex.quote` when available:

    try:
        from shlex import quote
    except ImportError:
        from pipes import quote